users/src/main/java/io/crate/operation/user/UserManagerService.java

```diff
@@ -61,6 +69,24 @@
     }
 
     public Iterable<User> userGetter() {
-        return ImmutableList.of(crateUser);
+        return users;
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (!event.metaDataChanged()) {
+            return;
+        }
+        users = getNewCurrentUsers(event.state().metaData().custom(UsersMetaData.TYPE));
+    }
+
+    static Set<User> getNewCurrentUsers(@Nullable UsersMetaData metaData) {
```

Comment:
this method name is a bit weird. current users, or new users?
I think something like `getUsersFromClusterState()` would be a better name